src/sentry/models/organization.py

```diff
@@ -279,7 +278,7 @@ def delete(self, **kwargs):
         # There is no foreign key relationship so we have to manually cascade.
         NotificationSetting.objects.remove_for_organization(self)
 
-        with transaction.atomic(), in_test_psql_role_override("postgres"):
+        with outbox_context(transaction.atomic(), kwargs, flush=False):
```

Comment:
Should `flush=False` be the default? Having potentially long synchronous operations throughout the application seems like it could increase fragility. We'll eventually have to relax many of these constraints. Should we take the risk now and add small async bumps while async is relatively 'fast' and enable writes as required?

Reply:
Hmmm.  Long synchronous operations?  to be clear, these outboxes are generally themselves single rpc calls.  when there is a backup of outboxes for a given shard (generally this is a problem for organization scope), it's still important in some sense to potentially have backpressure in the form of synchronous time waiting.

As far as we can see so far in production, these operations do not have high steady state volume -- rather, they can have high burst volume in the case of API interactions that may be processing several items at once.

It's arguable that some cases flushing is not necessary, but I'm curious to still try this.  I'm seeing more incidental pain from not processing outboxes synchronously effective provisioning flows, or outbox contention from lack of pessimistic locking.

I am willing to talk more about this.

Reply:
But yeah, I go back and forth.  I'm hoping there is some simple, obvious answer for behaviors.

Reply:
Long is relative here. I'm comparing the RPC duration to a DB query, and the RPC call will be 'longer'.

I agree there isn't a clear cut answer, but we have an opinion/stance (default to flush) and we can see how it plays out. If we're wrong pivoting in the future is low cost.